SQLException: Invalid character in connection string.

I just got the following exception when starting exasol-testcontainers:


```
java.sql.SQLException: Invalid character in connection string.

	at com.exasol.jdbc.EXADriver.parseURL8(EXADriver.java:570)
	at com.exasol.jdbc.EXADriver.connect(EXADriver.java:172)
	at com.exasol.containers.ExasolContainer.createConnectionForUser(ExasolContainer.java:335)
	at com.exasol.containers.ExasolContainer.createConnection(ExasolContainer.java:345)
```

on second run (with reuse) I got:
```
Caused by: org.testcontainers.containers.ContainerLaunchException: F-ETC-5: Exasol container start-up timed out trying connection to 'jdbc:exa:localhost/ec3be8628f3939cc979928b3304f598212bf05878e916c15b7c8c2694fa7fc50:49354;validateservercertificate=1' using query 'SELECT 1 FROM DUAL' after '362.2' seconds. Last connection exception was: 'Could not create new connection'
	at com.exasol.containers.ExasolContainer.waitUntilStatementCanBeExecuted(ExasolContainer.java:628)
	at com.exasol.containers.ExasolContainer.waitUntilContainerStarted(ExasolContainer.java:487)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:468)
```

--- a/doc/changes/changes_6.1.0.md
+++ b/doc/changes/changes_6.1.0.md
@@ -15,6 +15,7 @@ Cross-container `IMPORT` now also works with version 7.1 when using the TLS fing
 ## Bugfixes
 
 * #156: Fixed test case for cross-container import with Exasol 7.1
+* #180: Moved fingerprint to JDBC parameter for driver backward compatibility
 
 ## Dependency Updates
 

--- a/src/main/java/com/exasol/containers/ExasolContainer.java
+++ b/src/main/java/com/exasol/containers/ExasolContainer.java
@@ -253,8 +253,8 @@ public class ExasolContainer<T extends ExasolContainer<T>> extends JdbcDatabaseC
     }
 
     private String getJdbcUrlWithFingerprint(final String fingerprint) {
-        return "jdbc:exa:" + getContainerIpAddress() + "/" + fingerprint + ":" + getFirstMappedDatabasePort()
-                + ";validateservercertificate=1";
+        return "jdbc:exa:" + getContainerIpAddress() + ":" + getFirstMappedDatabasePort()
+                + ";validateservercertificate=1;fingerprint=" + fingerprint + ";";
     }
 
     private String getJdbcUrlWithoutFingerprint() {
@@ -892,11 +892,11 @@ public class ExasolContainer<T extends ExasolContainer<T>> extends JdbcDatabaseC
     }
 
     /**
-     * Get the SHA256 fingerprint of the TLS certificate used by the database in the container for database
-     * connections and the RPC interface.
+     * Get the SHA256 fingerprint of the TLS certificate used by the database in the container for database connections
+     * and the RPC interface.
      *
-     * @return SHA256 fingerprint of the TLS certificate or an empty {@link Optional} when the certificate file does
-     *         not exist.
+     * @return SHA256 fingerprint of the TLS certificate or an empty {@link Optional} when the certificate file does not
+     *         exist.
      */
     public Optional<String> getTlsCertificateFingerprint() {
         return this.certificateProvider.getSha256Fingerprint();

--- a/src/test/java/com/exasol/containers/ExasolContainerIT.java
+++ b/src/test/java/com/exasol/containers/ExasolContainerIT.java
@@ -42,8 +42,8 @@ class ExasolContainerIT {
         final String zeroTo255 = "([01]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])";
         final String ipRegexp = zeroTo255 + "\\." + zeroTo255 + "\\." + zeroTo255 + "\\." + zeroTo255;
         final String hostNamePattern = "(localhost|" + ipRegexp + ")";
-        assertThat(CONTAINER.getJdbcUrl(),
-                matchesPattern("jdbc:exa:" + hostNamePattern + "/\\w{64}:\\d{1,5};validateservercertificate=1"));
+        assertThat(CONTAINER.getJdbcUrl(), matchesPattern(
+                "jdbc:exa:" + hostNamePattern + ":\\d{1,5};validateservercertificate=1;fingerprint=\\w{64};"));
     }
 
     @Test

--- a/src/test/java/com/exasol/containers/ExasolContainerIT.java
+++ b/src/test/java/com/exasol/containers/ExasolContainerIT.java
@@ -48,7 +48,7 @@ class ExasolContainerIT {
 
     @Test
     void testJdbcUrlValidatesServerCertificate() throws SQLException {
-        assertThat(CONTAINER.getJdbcUrl(), endsWith(";validateservercertificate=1"));
+        assertThat(CONTAINER.getJdbcUrl(), containsString(";validateservercertificate=1"));
     }
 
     @Test

--- a/src/test/java/com/exasol/containers/tls/TlsConnectionIT.java
+++ b/src/test/java/com/exasol/containers/tls/TlsConnectionIT.java
@@ -96,7 +96,7 @@ class TlsConnectionIT {
     void testJdbcUrlContainsFingerprint() {
         final String jdbcUrl = CONTAINER.getJdbcUrl();
         final String expectedFingerprint = getFingerprint();
-        assertThat(jdbcUrl, containsString("/" + expectedFingerprint + ":"));
+        assertThat(jdbcUrl, containsString(";fingerprint=" + expectedFingerprint + ";"));
     }
 
     @Test

